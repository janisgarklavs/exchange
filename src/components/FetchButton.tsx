import React from 'react'

interface FetchButtonProps {
  loading: boolean
  clickHandler: () => void
}

export const FetchButton: React.SFC<FetchButtonProps> = ({ loading, clickHandler }) => (
  <button onClick={clickHandler} disabled={loading}>
    Get
  </button>
)
