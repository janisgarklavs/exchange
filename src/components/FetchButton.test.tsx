import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { FetchButton } from './FetchButton'

describe('<FetchButton />', () => {
  test('it renders correctly', () => {
    const clickHandler = jest.fn()
    const loading = false
    const component = shallow(<FetchButton loading={loading} clickHandler={clickHandler} />)
    expect(toJson(component)).toMatchSnapshot()
  })

  test('it calls function on button click', () => {
    const clickHandler = jest.fn()
    const loading = false
    const component = shallow(<FetchButton loading={loading} clickHandler={clickHandler} />)
    component.simulate('click')
    expect(clickHandler).toBeCalledTimes(1)
  })

  test('button is disabled while loading', () => {
    const clickHandler = jest.fn()
    const loading = true
    const component = shallow(<FetchButton loading={loading} clickHandler={clickHandler} />)
    expect(component.find('button').prop('disabled')).toBeTruthy()
  })
})
