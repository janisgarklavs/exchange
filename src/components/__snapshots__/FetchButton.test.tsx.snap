// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FetchButton /> it renders correctly 1`] = `
<button
  disabled={false}
  onClick={[MockFunction]}
>
  Get
</button>
`;
